plugins/UM3NetworkPrinting/src/Network/ZeroConfClient.py

```diff
@@ -129,20 +131,20 @@ def _onServiceAdded(self, zero_conf: Zeroconf, service_type: str, name: str) ->
 
         for record in zero_conf.cache.entries_with_name(info.server):
             info.update_record(zero_conf, time(), record)
-            if info.address:
+            if self._getAddress(info):
                 break
 
         # Request more data if info is not complete
-        if not info.address:
+        if not self._getAddress(info):
             new_info = zero_conf.get_service_info(service_type, name)
             if new_info is not None:
                 info = new_info
 
-        if info and info.address:
+        if info and self._getAddress(info):
             type_of_device = info.properties.get(b"type", None)
             if type_of_device:
                 if type_of_device == b"printer":
-                    address = '.'.join(map(str, info.address))
+                    address = '.'.join(map(str, self._getAddress(info)))
```

Comment:
I just noticed that I made it so that _getAddress() returns the address in byte form (or None) because it gets called four times and only once is the string actually needed. But when converting to a string here, the typing doesn't like the possibility of None being returned, even though that is ruled out by the test above. I haven't used the typing system enough to know what the best way to deal with this is.

Reply:
The problem here is that map expects an iterable, but _getAddress() returns Optional[bytes]. So if map sees None here, it will raise a 'NonType is not iterable' exception. Mypy complains because of that. So this case needs to be handled (by making next to not return None, and surrounding line 147 with a try-except StopIteration).

Also, I notice that `'.'.join(map(str, b"helloWorld"))` will output `'104.101.108.108.111.87.111.114.108.100'`. Are you sure this should return bytes? 
Note: I cannot get it to invoke this line of code locally, so I am asking to be sure.

Reply:
Even if _getAddress() raised StopIteration instead of returning None there, info.address on older zeroconf versions would still return None, and you would have to add a try-except to all the four places where _getAddress() is called. Instead I went with a list of addresses which still satisfies the boolean checks, we just need to take the first item when converting to a string here.

Zeroconf stores the addresses as bytes, so for example `'192.168.178.1'` would be `b'\xc0\xa8\xb2\x01'`. When iterating over a bytes object, the individual elements are represented as int, so this line converts the address bytes from _getAddress into its standard string representation. 

I have tested this with old and new zeroconf versions, with and without giving the ServiceInfo an address and all works as expected for me.